src/optable.ts

```diff
@@ -561,59 +578,90 @@ function frame_add_traceback(f: Py_FrameObject, t:Thread) {
     t.addToTraceback(tback);
 }
 
-// Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
-function fast_block_end (f: Py_FrameObject, t: Thread){
-    if (f.blockStack.length > 0) {
-        // Search for exception handler in current Py_FrameObject
-        find_exception_handler(f);
+// Search for exception handler in previous frames
+function find_in_prev(f: Py_FrameObject, t: Thread) {
+    frame_add_traceback(f, t);
+    while (f.stack.length > 0) {
+        f.pop();
     }
-    else {
-        // For the case where no exception handler exists in the current Py_FrameObject
-        // Unwinds the stack and moves to previous frames and searches for an exception handler
-        while (f.stack.length > 0) {
-            f.pop();
-        }
-       
-        var exception_handler_found: boolean = false;
-
-        // [CHANGE ONCE ARG>0 IMPLEMENTED (CO_VARGS)]incase of a raise with no arguements/no explicit catching exception
-        t.addToTraceback("TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType\n");
+     // Search for exception handler in previous frames
+    while (f.back != null) {
+        var chars = f.codeObj.lnotab.toString();
+        // stop frame execution
+        f.returnToThread = true;
+        f = (<Py_FrameObject>f.back);
+        // Add to traceback
         frame_add_traceback(f, t);
-
-         // Search for exception handler in previous frames
-        while (f.back != null) {
-            var chars = f.codeObj.lnotab.toString();
-            // stop frame execution
-            f.returnToThread = true;
-            f = (<Py_FrameObject>f.back);
-            // Add to traceback
-            frame_add_traceback(f, t);
-            // If an exception handler is found in either this frame or previous frames
-            if (find_exception_handler(f)) {
-                exception_handler_found = true;
-                break;
-            }
+        // If an exception handler is found in either this frame or previous frames
+        if (find_exception_handler(f)) {
+            return true;
         }
-        // Print traceback and cease execution
-        if (!exception_handler_found) {
-            t.writeTraceback();
+    }
+    // return false if no handler found
+    return false;
+}
+// Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+// TODO: Move logic to Thread.throwException
+function fast_block_end (f: Py_FrameObject, t: Thread){
```

Comment:
Add a return type annotation to this function signature, please. (Same goes for all the other functions without return types here.)